packages/p2p/source/peer-api-node-discoverer.ts

```diff
@@ -51,4 +59,24 @@ export class PeerApiNodeDiscoverer implements Contracts.P2P.PeerApiNodeDiscovere
 			),
 		);
 	}
+
+	async discoverNewApiNodes(): Promise<any> {
+		const peers = Utils.shuffle(this.peerRepository.getPeers()).slice(0, 5);
+		return Promise.all(peers.map((peer) => this.discoverApiNodes(peer)));
+	}
+
+	async refreshApiNodes(): Promise<any> {
+		return Promise.all(
+			this.apiNodeRepository
+				.getApiNodes()
+				.filter((apiNode) =>
+					(apiNode.lastPinged ?? dayjs()).isAfter(dayjs().add(randomNumber(10, 20), "minutes")),
```

Comment:
Since we call this method every 2 seconds, the chance node will be checked every 20 minutes is minimal. Almost every check will happen in 10th minute, because method will be called 30 times per minute. 

Suggestion: 
Calculate timeout only once and store it in variable. If timeout has passed refresh all the api nodes. Move that logic into Service. #checkApiNodes().